### ISSUE TITLE
Add various loss functions to the api

### ISSUE THREAD

**Issue by ManrajGrover** [tensorflow/tfjs-core#770]
Currently, the api only supports Softmax Cross Entropy. I feel we can other loss functions to the api. Listing them down here:

1. Absolute Difference
2. Cosine Distance
3. Hinge Loss
4. Huber Loss
5. Log Loss
6. Mean Pairwise Squared Error
7. Mean Squared Error
8. Sigmoid Cross Entropy Loss

I can work on these one by one.

**Comment by nsthorat** [comment]
Adding @caisq

**Comment by nsthorat** [comment]
This issue was moved to tensorflow/tfjs#131
